projects/ad3552r_fmcz/src/examples/axi_qspi/axi_qspi_example.c

```diff
@@ -0,0 +1,188 @@
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <inttypes.h>
+#include "common_data.h"
+#include "no_os_error.h"
+#include "no_os_print_log.h"
+#include "no_os_spi.h"
+#include "no_os_gpio.h"
+#include "no_os_util.h"
+#include "no_os_delay.h"
+#include "xilinx_spi.h"
+#include "xilinx_gpio.h"
+#include "ad3552r.h"
+
+#include <xparameters.h>
+#include <xil_cache.h>
+
+#define NUM_CYCLES 8
+
+extern const uint16_t no_os_sine_lut_16[512];
+
+int32_t init_gpios_to_defaults()
+{
+	const uint8_t gpios_initial_value[][2] = {
+		[GPIO_RESET_N] = {NO_OS_GPIO_OUT, NO_OS_GPIO_HIGH},
+		[GPIO_LDAC_N] = {NO_OS_GPIO_OUT, NO_OS_GPIO_HIGH},
+		[GPIO_SPI_QPI] = {NO_OS_GPIO_OUT, NO_OS_GPIO_LOW},
+		[GPIO_ALERT_N] = {NO_OS_GPIO_IN, 0},
+		[GPIO_9] = {NO_OS_GPIO_OUT, NO_OS_GPIO_HIGH},
+		[GPIO_RED] = {NO_OS_GPIO_OUT, NO_OS_GPIO_HIGH},
+		[GPIO_GREEN] = {NO_OS_GPIO_OUT, NO_OS_GPIO_HIGH},
+		[GPIO_BLUE] = {NO_OS_GPIO_OUT, NO_OS_GPIO_HIGH},
+	};
+	struct no_os_gpio_desc *gpio;
+	struct no_os_gpio_init_param param = default_gpio_param;
+	uint32_t i;
+	int32_t	 err;
+
+	for (i = 0; i < TOTAL_GPIOS; i++) {
+		param.number = GPIO_OFFSET + i;
+		err = no_os_gpio_get(&gpio, &param);
+		if (NO_OS_IS_ERR_VALUE(err))
+			return err;
+		if (gpios_initial_value[i][0] == NO_OS_GPIO_IN)
+			err = no_os_gpio_direction_input(gpio);
+		else
+			err = no_os_gpio_direction_output(gpio,
+							  gpios_initial_value[i][1]);
+
+		if (NO_OS_IS_ERR_VALUE(err))
+			return err;
+
+		no_os_gpio_remove(gpio);
+	}
+
+	return 0;
+}
+
+void set_power_up_success_led()
+{
+	struct no_os_gpio_desc *gpio;
+	struct no_os_gpio_init_param param = default_gpio_param;
+
+	param.number = GPIO_OFFSET + GPIO_GREEN;
+	no_os_gpio_get(&gpio, &param);
+	no_os_gpio_direction_output(gpio, NO_OS_GPIO_LOW);
+	no_os_gpio_remove(gpio);
+}
+
+int32_t run_example(struct ad3552r_desc *dac)
+{
+	int32_t err;
+	uint16_t samples[2] __attribute__ ((aligned));
+
+	samples[0] = 65534;
+	samples[1] = 0;
+
+	pr_info("Writing raw samples, ch0/1 %d/%d, using LDAC\n",
+		samples[0], samples[1]);
+
+	err = ad3552r_write_samples(dac, samples, 1,
+				    AD3552R_MASK_ALL_CH,
+				    AD3552R_WRITE_INPUT_REGS_AND_TRIGGER_LDAC);
+	if (err) {
+		pr_info("error writing samples\n");
+		return err;
+	}
+
+	no_os_mdelay(1000);
+
+	samples[0] = 0;
+	samples[1] = 65534;
+
+	pr_info("Writing raw samples, ch0/1 %d/%d, direct DAC REG write\n",
+		samples[0], samples[1]);
+
+	err = ad3552r_write_samples(dac, samples, 1,
+				    AD3552R_MASK_ALL_CH,
+				    AD3552R_WRITE_DAC_REGS);
+	if (err) {
+		pr_info("error writing samples\n");
+		return err;
+	}
+
+	no_os_mdelay(1000);
+
+	pr_info("Fast cyclic dma transfer starts now, for 20 seconds ...\n");
+
+	/* Use cyclic_secs to 0 for continuous cycling */
```

Comment:
hmmm i think this comment is a leftover

Reply:
It was not well formulated, adjusted the comment, can be useful